File: shapes polymorphism.py
import abc

class shape(metaclass=abc.ABCMeta):
    def __init__(self, name="null", length="null", width="null", base="null", height="null"):
        self._name = name
        self._length = length
        self._width = width
        self._height = height
        self._base = base

    @abc.abstractmethod
    def shapeData(self, store):
        pass

    @abc.abstractmethod
    def insert(self, shapename, store, index):
        pass

    # Getter and Setter for name
    def get_name(self):
        return self._name

    def set_name(self, name):
        self._name = name

    # Getter and Setter for length
    def get_length(self):
        return self._length

    def set_length(self, length):
        self._length = length

    # Getter and Setter for width
    def get_width(self):
        return self._width

    def set_width(self, width):
        self._width = width

    # Getter and Setter for height
    def get_height(self):
        return self._height

    def set_height(self, height):
        self._height = height

    # Getter and Setter for base
    def get_base(self):
        return self._base

    def set_base(self, base):
        self._base = base

class triangle(shape):
    def shapeData(self, store):
        print(f"Data Entered Successfully \n Your Data is:")
        for shapes in store:
            print(shapes)

    def insert(self, shapename, store, index):
        if index > 100:
            print("Shapes Linkedlist are full. Cannot add more Shapes.")
            return

        height = input("Enter Height : ")
        base = float(input("Enter Base : "))

        data = {
            "name": shapename,
            "height": height,
            "base": base
        }

        store.append(data)

        self.shapeData(store)

class square(shape):
    def shapeData(self, store):
        print(f"Data Entered Successfully \n Your Data is:")
        for shapes in store:
            print(shapes)

    def insert(self, shapename, store, index):
        if index > 100:
            print("Shapes Linkedlist are full. Cannot add more Shapes.")
            return

        length = float(input("Enter Length: "))

        data = {
            "name": shapename,
            "length": length
        }

        store.append(data)

        self.shapeData(store)

class rectangle(shape):
    def shapeData(self, store):
        print(f"Data Entered Successfully \n Your Data is:")
        for shapes in store:
            print(shapes)

    def insert(self, shapename, store, index):
        if index > 100:
            print("Shapes Linkedlist is full. Cannot add more Shapes.")
            return

        length = float(input("Enter Length: "))
        width = float(input("Enter Width: "))

        data = {
            "name": shapename,
            "length": length,
            "width": width
        }

        store.append(data)

        self.shapeData(store)

if __name__ == "__main__":
    store = []
    index = 0
    triangle_obj = triangle()
    square_obj = square()
    rectangle_obj = rectangle()
    for i in range(0, 100):
        try:
            index += 1
            shapename = input("Enter Shape Name: ").lower()  # just to make generic and simple code
            if shapename == "triangle":
                triangle_obj.insert(shapename, store, index)
            elif shapename == "square":
                square_obj.insert(shapename, store, index)
            elif shapename == "rectangle":
                rectangle_obj.insert(shapename, store, index)
            else:
                msg = "Unknown Type Error"
                alert = TypeError(msg)
                print(alert)
        except:
            msg = "Invalid Type Error"
            TypeError(msg)

File: shapes polymorphism.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

abstract class Shape {
    protected String name;
    protected double length;
    protected double width;
    protected double height;
    protected double base;

    public Shape() {
        this.name = "null";
        this.length = -1.0;
        this.width = -1.0;
        this.height = -1.0;
        this.base = -1.0;
    }

    public abstract void shapeData(List<String> store);

    public abstract void insert(String shapename, List<String> store, int index);

    // Getters and Setters
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public double getLength() {
        return length;
    }

    public void setLength(double length) {
        this.length = length;
    }

    public double getWidth() {
        return width;
    }

    public void setWidth(double width) {
        this.width = width;
    }

    public double getHeight() {
        return height;
    }

    public void setHeight(double height) {
        this.height = height;
    }

    public double getBase() {
        return base;
    }

    public void setBase(double base) {
        this.base = base;
    }
}

class Triangle extends Shape {
    @Override
    public void shapeData(List<String> store) {
        System.out.println("Data Entered Successfully");
        System.out.println("Your Data is:");
        for (String shape : store) {
            System.out.println(shape);
        }
    }

    @Override
    public void insert(String shapename, List<String> store, int index) {
        if (index > 100) {
            System.out.println("Shapes Linkedlist is full. Cannot add more Shapes.");
            return;
        }

        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter Height: ");
        double height = scanner.nextDouble();
        System.out.print("Enter Base: ");
        double base = scanner.nextDouble();

        String data = "name: " + shapename + ", height: " + height + ", base: " + base;
        store.add(data);

        shapeData(store);
    }
}

class Square extends Shape {
    @Override
    public void shapeData(List<String> store) {
        System.out.println("Data Entered Successfully");
        System.out.println("Your Data is:");
        for (String shape : store) {
            System.out.println(shape);
        }
    }

    @Override
    public void insert(String shapename, List<String> store, int index) {
        if (index > 100) {
            System.out.println("Shapes Linkedlist is full. Cannot add more Shapes.");
            return;
        }

        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter Length: ");
        double length = scanner.nextDouble();

        String data = "name: " + shapename + ", length: " + length;
        store.add(data);

        shapeData(store);
    }
}

class Rectangle extends Shape {
    @Override
    public void shapeData(List<String> store) {
        System.out.println("Data Entered Successfully");
        System.out.println("Your Data is:");
        for (String shape : store) {
            System.out.println(shape);
        }
    }

    @Override
    public void insert(String shapename, List<String> store, int index) {
        if (index > 100) {
            System.out.println("Shapes Linkedlist is full. Cannot add more Shapes.");
            return;
        }

        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter Length: ");
        double length = scanner.nextDouble();
        System.out.print("Enter Width: ");
        double width = scanner.nextDouble();

        String data = "name: " + shapename + ", length: " + length + ", width: " + width;
        store.add(data);

        shapeData(store);
    }
}

public class Main {
    public static void main(String[] args) {
        List<String> store = new ArrayList<>();
        int index = 0;
        Triangle triangleObj = new Triangle();
        Square squareObj = new Square();
        Rectangle rectangleObj = new Rectangle();

        for (int i = 0; i < 100; i++) {
            try {
                index++;
                Scanner scanner = new Scanner(System.in);
                System.out.print("Enter Shape Name: ");
                String shapename = scanner.nextLine().toLowerCase();

                if (shapename.equals("triangle")) {
                    triangleObj.insert(shapename, store, index);
                } else if (shapename.equals("square")) {
                    squareObj.insert(shapename, store, index);
                } else if (shapename.equals("rectangle")) {
                    rectangleObj.insert(shapename, store, index);
                } else {
                    System.out.println("Unknown Type Error");
                }
            } catch (Exception e) {
                System.out.println("Invalid Type Error");
            }
        }
    }
}
